src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -47,11 +47,23 @@ pub struct PlanAggCall {
     pub distinct: bool,
 }
 
+struct InputRefWithDistinct<'a>(&'a InputRef, bool);
+
+impl<'a> fmt::Debug for InputRefWithDistinct<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.1 {
+            write!(f, "distinct ")?;
+        }
+        write!(f, "{}", self.0)
+    }
+}
+
 impl fmt::Debug for PlanAggCall {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         let mut builder = f.debug_tuple(&format!("{}", self.agg_kind));
         self.inputs.iter().for_each(|child| {
-            builder.field(child);
+            let input_ref = InputRefWithDistinct(child, self.distinct);
```

Comment:
Based on the fact that `distinct` is a variable independent of `inputRef` in `PlanAggCall`, we will not have the case that we have multiple input refs but only some of them is `distinct`?

Reply:
Yes, we do not [foresee](https://www.postgresql.org/docs/current/sql-expressions.html#:~:text=or%20distinct%20set%20of%20values%2C%20for%20multiple%20expressions) any agg call to be distinct only on part of its inputs.

However, when an agg call has multiple inputs (e.g. `string_agg`), it does not look well: `string_agg(distinct $0, distinct $1)`.

Alternatively we can use `string_agg-distinct($0, $1)`, or avoid `debug_tuple` and implement our own formatting to get `string_agg(distinct $0, $1)`.

Reply:
I implemented our own formatting